### DISCUSSION TITLE
Welcome to grammars-v4 Discussions!

### DISCUSSION THREAD

**Discussion by parrt** [antlr/grammars-v4#2845] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 7020232604** [comment]
i am getting below error ------------->>> 'PlSqlParser' object has no attribute 'parse'  can anyone help me to resolve this issue?

**Comment by gurucubano** [comment]
Hi there,
My name is Matthias. I'm relatively new to Java and absolute new to ANTLR. My background is 45 years of UNIX, from Amdahl UTS to FreeBSD, C, shell, PostgreSQL, .... More here: http://www.unixarea.de/
My goal for which I came to here is port a grammar written in an old web application, not by me and 20 years ago, from ANTLR2+AST to ANTLR4. Thanks for your help and hints in advance.
Matthias

**Reply by kaby76** [reply]
You'll have to extract the EBNF first from your grammar. And, you'll have to decide how you want to port tree construction and semantics involved in parsing. Antlr4 does not support tree construction operators anymore.

There isn't a converter to extract the EBNF from an Antlr2 grammar. You could try to use the [antlr2 grammar](https://github.com/antlr/grammars-v4/tree/master/antlr/antlr2) to parse the input and use [Trash](https://github.com/kaby76/Trash) to rip out non-EBNF content.

**Reply by gurucubano** [reply]
I did:
```
git clone https://github.com/antlr/grammars-v4.git
cd grammars-v4/antlr/antlr2
mvn clean
mvn install
ls -l target/antlr2-1.0-SNAPSHOT.jar
-rw-r--r--  1 apitzm  32873736  111737 15 Apr 12:54 target/antlr2-1.0-SNAPSHOT.jar
alias | grep grun
alias grun='java -Xmx500M -cp "/usr/local/lib/antlr-4.13.2-complete.jar:$CLASSPATH" org.antlr.v4.gui.TestRig'
export CLASSPATH=$CLASSPATH:target/antlr2-1.0-SNAPSHOT.jar
```
and now I don't know to run it to parse the ANTLR2 grammar file `InfoGuideQuery.g`

In general, all the example code project lack somehow a README file which describes the steps to get them executed; it took me some time to figure out the first one `grammars-v4/useragent` as:
```
git clone https://github.com/antlr/grammars-v4.git
cd grammars-v4/useragent
mvn clean
mvn install
alias grun='java -Xmx500M -cp "/usr/local/lib/antlr-4.13.2-complete.jar:$CLASSPATH" org.antlr.v4.gui.TestRig'
export CLASSPATH=${CLASSPATH}:target/useragent-1.0-SNAPSHOT.jar
 
cat examples/example1.txt
Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405
 
grun useragent prog -tree < examples/example1.txt
 
(prog (product (name Mozilla) / (version 5 . 0)) (comment (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us)) (product (name AppleWebKit) / (version 531 . 21 . 10)) (comment (KHTML, like Gecko)) (product (name Mobile) / (version 7B405)) <EOF>)
```

**Reply by kaby76** [reply]
We don't put the steps on how to build and test the grammar in the readme because we would have to do that for 350 grammars _times_ 8 target languages _times_ 3 different OSes (Windows is not Unix) _times_ 2 different generators ("Official Antlr" and now Antlr-ng), and update that with every release of every code generator. Instead, we should add your Java-specific instructions to the repo wiki. https://github.com/antlr/grammars-v4/wiki#how-do-i-test-the-grammars

Take care on mixing generators and runtime of different versions. This is because the old "Maven tester" uses Antlr 4.11.1 to generate the .java files, whereas you link in the Antlr4 runtime using 4.13.2. Generally, you should never do that because of feature differences. The desc.xml may indicate what minimum version of Antlr4 is required for a grammar.

**Reply by gurucubano** [reply]
> Instead, we should add your Java-specific instructions to the repo wiki. https://github.com/antlr/grammars-v4/wiki#how-do-i-test-the-grammars

Feel free to copy my example given for grammars-v4/useragent.

**Comment by donreamey** [comment]
Hi I'm Don.

I'm doing some work to migrate metadata from different SQL languages to Microsoft T-SQL.

I'm the Software Engineering Manager for [Microsoft Fabric Migration Assistant](https://learn.microsoft.com/en-us/fabric/data-warehouse/migration-assistant)

I'm currently making some changes to the Snowflake grammar and I would like permissions to push to my branch and create PR for review.

Thanks,
Don

**Reply by kaby76** [reply]
Hi Don. Sounds fine. Please cc @mlorek on your PR so he can review. He has written much of the grammar and made some changes recently. Ken

**Reply by donreamey** [reply]
hi @kaby76, @mlorek 

I don't have permission to push to my dev branch. Do you know who can give permissions to push to a dev branch?

Don

```
Q:\Repos\grammars-v4>git push --set-upstream origin dev/dreamey/parsestatementbody
remote: Permission to antlr/grammars-v4.git denied to donreamey.
fatal: unable to access 'https://github.com/antlr/grammars-v4.git/': The requested URL returned error: 403
```

**Reply by kaby76** [reply]
You are trying to modify the repo directly. You can't.

You need to fork the repo first (in a browser), clone your copy of grammars-v4 ("git clone https://github.com/donreamey/grammars-v4.git"), modify that, git commit, then git push to your copy of the repo.

 Once you have checked in you changes, you can create a PR (in a browser, while browsing https://github.com/antlr/grammars-v4, go to "New" PR), then set it up to create the PR from your cloned copy back to grammars-v4.

**Reply by donreamey** [reply]
Thanks. I'm so used to creating a dev branch I forgot about forking first.
